Robot/MistyCharacter/Character/AnimationManager.cs

```diff
@@ -689,16 +691,23 @@ private async Task<bool> WaitOnSpeechCompletionEvent(int timeoutMs)
 			return response;
 		}
 
-		private double? GetNullableObject(string [] array, int index)
+		private double? GetNullableObject(string [] array, int index, bool treatEmptyAsNull = false)
 		{
-			if (array == null || array.Length < index || array.ElementAtOrDefault(index) == null || array.ElementAtOrDefault(index).Trim() == "null")
+			if (array == null || array.Length < index || array.ElementAtOrDefault(index) == null || array.ElementAtOrDefault(index).Trim().ToLower() == "null" ||
+				(treatEmptyAsNull && string.IsNullOrWhiteSpace(array.ElementAtOrDefault(index))))
 			{
 				return null;
 			}
 			return Convert.ToDouble(array[index]);
 		}
 
-
+		private async Task ManageJavaScriptRecording(string command)
+		{
+			if (!string.IsNullOrWhiteSpace(command) && _javaScriptRecorder != null && CharacterParameters.CreateJavaScriptTemplate)
+			{
+				await _javaScriptRecorder.SaveRecording(command);
+			}
+		}
 
 		private async Task<CommandResult> ProcessCommand(string command, bool guaranteedCommand, int loop)

```

Comment:
Now really isn't the time, but you may want to see if a command/handler sort of pattern is a better fit here. It seems like there may be a logical set of interfaces that can manage this work.

Reply:
Yeah, there are a number of things that I want to rework now that the flow of the system has changed so much, but wanted to get this functionality out first as it has some needed features and bug fixes.